packages/core/src/collections/trash.ts

```diff
@@ -192,6 +197,12 @@ export default class Trash {
     }
 
     if (notebookIds.length > 0) {
+      const canRestoreNotebooks =
+        (await this.db.notebooks.all.count()) >= FREE_NOTEBOOKS_LIMIT &&
```

Comment:
This'll only work prevent restoration if the notebooks length exactly matches `FREE_NOTEBOOKS_LIMIT` but what if user has 10 deleted notebooks in trash and 12 notebooks otherwise? After restoration, the user will have a total of 22 notebooks (2 more than allowed).

Reply:
Nice catch! I've included `notebookIds.length` in the check, hope that caters to this case